src/main/java/com/example/spring/models/ShoppingCart.java

```diff
@@ -0,0 +1,36 @@
+package com.example.spring.models;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.Fetch;
+import org.hibernate.annotations.FetchMode;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Table(name = "shopping_carts")
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = TRUE WHERE id = ?")
+@SQLRestriction(value = "is_deleted = FALSE")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne
```

Comment:
You can also add `@MapsId`

Reply:
after adding @MapsId I receive a wrong result for getCart method 
<img width="1440" alt="Screenshot 2024-09-20 at 16 15 39" src="https://github.com/user-attachments/assets/a0db3f9f-789a-40a6-9824-e8e5ca0086c8">
I guess, this happens so, because in my table there are 2 different columns is and user_id, and if I add @MapsId, field 'id' gets the same value like field 'userId', but in table 'shopping_carts' I don't have a shopping cart with id = 6

Reply:
yes, this is probably the reason, create a new user, add products to them, and check how it works with a new user

Reply:
nothing changed, maybe, I have to change a structure of table 'shopping cart', it will be have only one column and then it will work